app/models/accounting/transaction.rb

```diff
@@ -2,37 +2,50 @@
 #
 # Table name: accounting_transactions
 #
-#  created_at          :datetime         not null
-#  id                  :integer          not null, primary key
-#  qb_transaction_id   :string           not null
-#  qb_transaction_type :string           not null
-#  updated_at          :datetime         not null
+#  accounting_account_id :integer
+#  created_at            :datetime         not null
+#  id                    :integer          not null, primary key
+#  project_id            :integer
+#  qb_id                 :string           not null
+#  qb_transaction_type   :string           not null
+#  quickbooks_data       :json
+#  updated_at            :datetime         not null
 #
 # Indexes
 #
-#  acc_trans_qbid_qbtype_unq_idx                         (qb_transaction_id,qb_transaction_type) UNIQUE
-#  index_accounting_transactions_on_qb_transaction_id    (qb_transaction_id)
-#  index_accounting_transactions_on_qb_transaction_type  (qb_transaction_type)
+#  acc_trans_qbid_qbtype__unq_idx                          (qb_id,qb_transaction_type) UNIQUE
+#  index_accounting_transactions_on_accounting_account_id  (accounting_account_id)
+#  index_accounting_transactions_on_project_id             (project_id)
+#  index_accounting_transactions_on_qb_id                  (qb_id)
+#  index_accounting_transactions_on_qb_transaction_type    (qb_transaction_type)
+#
+# Foreign Keys
+#
+#  fk_rails_3b7e4ae807  (accounting_account_id => accounting_accounts.id)
+#  fk_rails_662fd2ba2d  (project_id => projects.id)
 #
 
 class Accounting::Transaction < ActiveRecord::Base
   TRANSACTION_TYPES = %w(JournalEntry Deposit Purchase).freeze
 
-  attr_writer :qb_object
+  belongs_to :account, inverse_of: :transactions, foreign_key: :accounting_account_id
+  belongs_to :project, inverse_of: :transactions, foreign_key: :accounting_account_id
 
-  delegate :txn_date, :total, :private_note, to: :qb_object
+  delegate :txn_date, :total, :private_note, to: :quickbooks_data
 
-  # Make sure to append this after the last link in the relation chain. It will ensure
-  # the quickbooks data is populated.
-  def self.with_qb_objects
-    transactions = all
-    return transactions if transactions.count < 1
+  def txn_date
+    quickbooks_data[:txn_date]
+  end
+
+  def total
+    quickbooks_data[:total]
+  end
 
-    Accounting::Quickbooks::Fetcher.new(transactions).fetch
+  def private_note
+    quickbooks_data[:private_note]
   end
 
-  def qb_object
-    raise ArgumentError, 'qb_object not already set. Make sure to append #with_qb_objects at the end of an AR chain.' if @qb_object.blank?
-    @qb_object
+  def quickbooks_data
+    super.with_indifferent_access
```

Comment:
Should still be fine, I didn't override `def quickbooks_data=`. I suppose you couldn't update the field without replacing the whole block of json. I doubt it will be an issue. We really don't want anyone besides quickbooks changing the data here.